File: src/main/java/leetCode/offer/Offer4.java
package leetCode.offer;

/**
 * @author jingxinwu
 * @date 2021-11-19 9:07 下午
 */
public class Offer4 {

    public static void main(String[] args) {
        Offer4 offer4 = new Offer4();
        int[][] arr = new int[][]{{-5}};
        System.out.println(offer4.findNumberIn2DArray(arr, -5));
    }


    public boolean findNumberIn2DArray(int[][] matrix, int target) {
        if (matrix == null || matrix.length == 0) {
            return false;
        }

        int i = matrix[0].length - 1;
        int j = 0;
        while (i >= 0 && j < matrix.length) {
            int temp = matrix[j][i];
            if (temp == target) {
                return true;
            } else if (temp < target) {
                j++;
            } else {
                i--;
            }
        }
        return false;
    }
}

File: src/main/GO/leetcode/lists/offer/Offer4.go
package offer

/**
 * @author wujingxinit@outlook.com
 * @date 2022/12/28 14:52
 */

func findNumberIn2DArray(matrix [][]int, target int) bool {
	if len(matrix) == 0 {
		return false
	}
	m, n := len(matrix), len(matrix[0])
	row, col := 0, n-1

	for row < m-1 && col >= 0 {
		if matrix[row][col] == target {
			return true
		}
		if matrix[row][col] < target {
			row++
		} else {
			col--
		}
	}
	return false
}
